### PR TITLE
game menu node 1, build system node 2

### PR DESCRIPTION
**Pull request by pavelskipenes**
- add can init
- add mcp2515 & can driver
- add spi
- can: pre rebase
- fix spi, mcp2515, can
- can testing
- can testing
- can
- can testing
- fix can
- can/mcp2515
- split mcp2515/can
- fix can
- fix can timing
- init-node2
- Fixed bit shifting for bit_timing in CAN_BR
- fix can timing
- testing can_send and can_recieve
- can/node1: disable interrupts and message reception
- can/node1: PS1:7,PS2:6,Prop:2,BRP:16
- node_2: fix build system
- formatting
- node_2: init can
- add can init
- node1/can: fix some compiler warnings
- node_2/Makefile: less verbose flashing
- node1/can: documentation, split public/private interface
- typo
- wip/node2: PWM driver
- add main menu and fix some bugs
- node2: fix build system, reorganize

### CODE REVIEW

File: nodes/node_1/src/game.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            oled_home(next_state);
+            break;
+        case SELECT:
+            if (!can_change)
+            {
+                break;
+            }
+            can_change = false;
+            printf("SELECT\n");
+            return next_state;
```
Comment (evenkjaexli) [inline nodes/node_1/src/game.c]: unused?

File: nodes/node_1/src/game.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+enum user_choice
```
Comment (evenkjaexli) [inline nodes/node_1/src/game.h]: move header file to /include

File: nodes/node_1/src/modules/include/menu.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+void menu_main_print_main_menu();
```
Comment (evenkjaexli) [inline nodes/node_1/src/modules/include/menu.h]:
```suggestion
void menu_main_print_main_menu(FILE *);
```

File: nodes/node_1/src/game.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            break;
+        }
+    }
+
+    return next_state;
+}
+
+enum user_choice wait_for_user_input()
+{
+    while (true)
```
Comment (KristianKristian) [inline nodes/node_1/src/game.c]: Why looping?

Comment (pavelskipenes) [reply nodes/node_1/src/game.c]: Yeah, that does not make sence

File: nodes/node_1/src/game.c (hunk 1 above)
Comment (pavelskipenes) [reply nodes/node_1/src/game.c]: yep

**Pull request merged**